packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -236,9 +246,47 @@ public ResponseEntity<UUID> equationsToModel(
 			}
 		}
 
+		// Cleanup equations from the request
+		List<String> equations = new ArrayList<>();
+		if (req.get("equations") != null) {
+			for (final JsonNode equation : req.get("equations")) {
+				equations.add(equation.asText());
+			}
+		}
+		TaskRequest cleanupReq = cleanupEquationsTaskRequest(projectId, equations);
+		final TaskResponse cleanupResp;
+		try {
+			cleanupResp = taskService.runTask(TaskMode.SYNC, cleanupReq);
+		} catch (final JsonProcessingException e) {
+			log.error("Unable to serialize input", e);
+			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, messages.get("task.gollm.json-processing"));
+		} catch (final TimeoutException e) {
+			log.warn("Timeout while waiting for task response", e);
+			throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, messages.get("task.gollm.timeout"));
+		} catch (final InterruptedException e) {
+			log.warn("Interrupted while waiting for task response", e);
+			throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, messages.get("task.gollm.interrupted"));
+		} catch (final ExecutionException e) {
+			log.error("Error while waiting for task response", e);
+			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, messages.get("task.gollm.execution-failure"));
+		}
+
+		// create a clone of the request and set the cleaned up equations
```

Comment:
Why do we need a clone here?

Reply:
so that I don't modify the original request.